build_tools/aws-sdk-code-generator/spec/interfaces/protocols_spec.rb

```diff
@@ -0,0 +1,90 @@
+# frozen_string_literal: true
+
+require_relative '../spec_helper'
+
+describe 'Protocols Resolution:' do
+  it 'raises on fake protocols' do
+    expect {
+      SpecHelper.generate_service(['ProtocolsFake'], multiple_files: false)
+    }.to raise_error(/unsupported protocol/)
+  end
+
+  context 'with smithy rpc protocol support' do
+    it 'selects smithy rpc v2 with priority' do
+      SpecHelper.generate_service(['ProtocolsRpcJson'], multiple_files: false)
+      client = ProtocolsRpcJson::Client.new(stub_responses: true)
+      expect(client.handlers).to include(Aws::RpcV2::Handler)
+      Object.send(:remove_const, :ProtocolsRpcJson)
+    end
+
+    it 'selects smithy rpc v2 if it is the only choice' do
+      SpecHelper.generate_service(['ProtocolsRpc'], multiple_files: false)
+      client = ProtocolsRpc::Client.new(stub_responses: true)
+      expect(client.handlers).to include(Aws::RpcV2::Handler)
+      Object.send(:remove_const, :ProtocolsRpc)
+    end
+
+    it 'selects smithy rpc v2 out of a list' do
+      SpecHelper.generate_service(['ProtocolsRpcJsonQuery'], multiple_files: false)
+      client = ProtocolsRpcJsonQuery::Client.new(stub_responses: true)
+      expect(client.handlers).to include(Aws::RpcV2::Handler)
+      Object.send(:remove_const, :ProtocolsRpcJsonQuery)
+    end
+
+    it 'selects json as fallback if smithy rpc v2 is not modeled' do
+      SpecHelper.generate_service(['ProtocolsJsonQuery'], multiple_files: false)
+      client = ProtocolsJsonQuery::Client.new(stub_responses: true)
+      expect(client.handlers).to include(Aws::Json::Handler)
+      Object.send(:remove_const, :ProtocolsJsonQuery)
+    end
+
+    it 'selects query if is the only choice' do
```

Comment:
I think you meant to say, `selects query if it is the only choice` here like line 20